### ISSUE TITLE
Problem building with --prod and --build-optimize wint ng-cli 1.3.0

### ISSUE THREAD

**Issue by reyiyo** [zefoy/ngx-perfect-scrollbar#70]
According to this issue: https://github.com/angular/angular-cli/issues/7110 it is because of the rxjs imports.

```
ERROR in ./node_modules/rxjs/observable/BoundCallbackObservable.js
Module build failed: TypeError: Cannot read property 'type' of undefined
    at Object.getEffectiveTypeAnnotationNode (...node_modules/typescript/lib/typescript.js:9341:17)
    at assignContextualParameterTypes (...node_modules/typescript/lib/typescript.js:41652:25)
    at checkFunctionExpressionOrObjectLiteralMethod (...node_modules/typescript/lib/typescript.js:41948:29)
    at checkExpressionWorker ...node_modules/typescript/lib/typescript.js:42959:28)
    at checkExpression (...node_modules/typescript/lib/typescript.js:42898:42)
    at checkExpressionCached (...node_modules/typescript/lib/typescript.js:42779:38)
    at checkReturnStatement (...node_modules/typescript/lib/typescript.js:45418:54)
    at checkSourceElement (...node_modules/typescript/lib/typescript.js:46763:28)
    at Object.forEach (...node_modules/typescript/lib/typescript.js:1506:30)
    at checkBlock (...node_modules/typescript/lib/typescript.js:44563:16)
    at checkSourceElement (...node_modules/typescript/lib/typescript.js:46742:28)
    at checkFunctionExpressionOrObjectLiteralMethodDeferred (...node_modules/typescript/lib/typescript.js:41990:21)
    at checkDeferredNodes (...node_modules/typescript/lib/typescript.js:46828:25)
    at checkSourceFileWorker (...node_modules/typescript/lib/typescript.js:46863:17)
    at checkSourceFile (...node_modules/typescript/lib/typescript.js:46842:13)
    at Object.forEach (...node_modules/typescript/lib/typescript.js:1506:30)
 @ ./node_modules/rxjs/observable/bindCallback.js 2:32-68
 @ ./node_modules/rxjs/add/observable/bindCallback.js
 @ ./node_modules/rxjs/Rx.js
 @ ./node_modules/ngx-perfect-scrollbar/dist/lib/perfect-scrollbar.component.js
 @ ./node_modules/ngx-perfect-scrollbar/dist/lib/perfect-scrollbar.module.js
 @ ./src/$$_gendir/app/app.module.ngfactory.ts
 @ ./src/main.ts
```

**Comment by sconix** [comment]
We are using angular-cli 1.3.0 and there is no problems with this library.

**Comment by sconix** [comment]
Ah it needs build optimizer usage, well to me this looks like a workaround so I am going to wait a bit to see a proper fix for this. Thanks for pointing this out and linking the issue, will monitor it and if the suggested fix will be really needed I will merge your PR.

**Comment by Franky238** [comment]
+1 here `    "ngx-perfect-scrollbar": "4.5.3",`

**Comment by sconix** [comment]
This seems to be a build optimizer bug, they have already closed the bug which indicates that it will be fixed with angular-cli 1.4.0. So I am waiting to see if it fixes this.

**Comment by sconix** [comment]
I will merge your PR since its good change anyway. If it also fixes this then all is good.
